File: piper/android/app/src/main/java/io/gopiper/piper/engine/statements/text_isEmpty.kt
package io.gopiper.piper.engine.statements

import android.util.Log
import io.gopiper.piper.engine.Piper
import org.json.JSONObject
import java.lang.Exception

suspend fun Piper.executeTextIsEmpty(statement: JSONObject, onError: (e: Exception) -> Unit): Boolean? {
    try {
        Log.d("hello", "execute_text_isEmpty")
        val pointer = statement["pointer"].toString()
        val text = executeStatement(pointer)?.toString()
        return text?.isEmpty()
    } catch (e: Exception) {
        onError(e)
        return null
    }
}

File: piper/extensions/chrome/src/engine/statements/text_isEmpty.js
import { stringify } from "../util"


export default async function(statement, onError) {
    try {
        let pointer = await this.executeStatement(stringify(statement.pointer))
        let text = stringify(pointer)
        return text.length === 0
    } catch (e) {
        onError(e)
    }
}
